packages/datagateway-download/src/downloadApi.ts

```diff
@@ -362,3 +362,28 @@ export const getDataUrl = (
     readSciGatewayToken().sessionId
   }&preparedId=${preparedId}&outname=${fileName}`;
 };
+
+/**
+ * Describes the progress of a download. Can be a percentage between 0-100
+ * or a string describing the status of the download.
+ */
+export type DownloadProgress = number | string;
+
+/**
+ * Return a percentage of files that have been restored for the given prepared ID.
+ * This will normally be an integer value between 0 and 100 but can also be a status value such as "UNKNOWN"
+ */
+export const getPercentageComplete = async ({
+  preparedId,
+  settings: { idsUrl },
+}: {
+  preparedId: string;
+  settings: { idsUrl: string };
+}): Promise<DownloadProgress> => {
+  const { data } = await axios.get(`${idsUrl}/getPercentageComplete`, {
+    params: { preparedId },
+  });
+  // if data is not a number (NaN), it is a status value
+  const isStatus = isNaN(data);
```

Comment:
I think `isNaN` is a bit unreliable - it has some odd results (`isNaN("")` is false, so is `isNaN([])`. I think the below code is more robust.

```
const numberData = parseInt(data); // or parseFloat - does the endpoint only return integers?
return Number.isNaN(numberData) ? data : numberData;
```

Reply:
I can use `parseFloat` for now, it is easy to switch to `parseInt` if needed.